Automattic/jetpack
# Social: Fix frontend media validation logic

Fixes https://github.com/Automattic/jpop-issues/issues/9123

This issues is that we are validating the featured image when the validation is not necessary, because the image won't get uploaded anywhere. The only edge case is Instagram

## Proposed changes:

* Put back the logic removed in #37964, but with the difference of using `attachedMedia.length` instead of the removed `shouldUploadAttachedMedia`

### Other information:


## Jetpack product discussion


https://github.com/Automattic/jpop-issues/issues/9123
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






* Have at least a Facebook and Instagram connection on your site, and have auto-conversion off.
* Create a new post and add a featured image that is bigger than 10Mb. On trunk this scenario should result in both Instagram and Facebook being disabled, because of the media restrictions.
* With this change only Instagram should get disabled, but not Facebook, because we only validate the featured image for Instagram.
* Test with auto-conversion on, it should still work as before
